[node_backup] Add support for other s3 providers

blakebyrnes opened the pull request:
You can configure another s3 provider by supplying the node_backup_s3_provider, node_backup_s3_access_key_id, node_backup_s3_secret_access_key, node_backup_s3_endpoint and optionally node_backup_s3_region. To use this type of service, you have to set the backup type to s3-rclone.

This change defaults to the previous configs (`node_backup_r2_access_key_id`, etc and a Cloudflare s3 provider.

blakebyrnes committed:
Add support for other s3 providers

You can configure another s3 provider by supplying the node_backup_s3_provider, node_backup_s3_access_key_id, node_backup_s3_secret_access_key, node_backup_s3_endpoint and optionally node_backup_s3_region. To use this type of service, you have to set the backup type to s3-rclone.

nprt commented on roles/node_backup/templates/rclone/rclone.conf.j2:
```diff
@@ -1,14 +1,18 @@
 upload_cutoff = 1024M
 upload_concurrency = {{ node_backup_max_concurrent_requests }}
 chunk_size = 256M
+no_check_bucket = true
```
I think we shouldn't disallow checks here, i.e. leave it as it is. It should be safer to keep the default option.

blakebyrnes replied:
This was the feature needed to get cloudflare working with rclone. Was a bit tricky to track down. We could only add it if provider is cloudflare, but I don't know what other providers need it

nprt replied:
Ah, got it! I think it might be better to add it only for cloudflare initially, and we can add it for the new ones, if a need arises. Do you think that makes sense?

blakebyrnes replied:
I agree. I think this feature isn't a security thing, it's just an optimization on how it checks before writing. That said, it seems better to only add if needed.

nprt commented on roles/node_backup/templates/single-backup.sh.j2:
```diff
@@ -128,7 +128,9 @@ echo -e "\n---\n$(date +%Y-%m-%d\ %H:%M:%S) Start the '{{ item.id }}' backup\n--
 remote="GCPbackups"
 {% elif item.type == 'r2-rclone' %}
-remote="R2backups"
+remote="S3backups"
```
I don't think we need to change this. The `s3-rclone` item.type should suffice for any new buckets. Also, we might break backwards compatibility if we change it.

blakebyrnes replied:
R2 wasn't working.. so probably not a backwards compatibility concern :) Do you think we should just yank the cloudflare specific config in general?

nprt replied:
Hah, yeah, you're right. :)
Not sure about yanking the cloudflare specific config. Maybe leave it like this, and come back to it later? Leaving it as is looks good to me for now.

nprt reviewed (commented):
Hey @blakebyrnes, thanks again for taking the time to submit this PR! 

I left a couple of comments. I'd love to hear your thoughts.

nprt commented:
> Hey @blakebyrnes, thanks again for taking the time to submit this PR!
> 
> I left a couple of comments. I'd love to hear your thoughts.

One last thing, could you, please, also change the version here: https://github.com/paritytech/ansible-polkadot/blob/main/galaxy.yml#L11? Once we bump the version there, it should publish the new release on Ansible Galaxy. 

We'll look into automating this in the future.

blakebyrnes committed:
Node backups - only use no_check_bucket on R2

Restrict the no_check_bucket to Cloudflare as the impact on other sites is unknown.

Also bumps galaxy.yml

nprt reviewed (approved):
Awesome, lgtm! Thank you so much for taking time to create the PR! 

If you could share a bit more on how you use this, and what else could help you in your day-to-day, or in general, I'd much appreciate it.

BulatSaif commented:
Thank you for contribution :+1:
